tmk_core/common/chibios/eeprom_teensy.c

```diff
@@ -231,18 +229,23 @@ void eeprom_initialize(void) {
         uint8_t stat = FTFL->FSTAT & 0x70;
         if (stat) FTFL->FSTAT = stat;
 
-        // FlexRAM is configured as traditional RAM
-        // We need to reconfigure for EEPROM usage
+            // FlexRAM is configured as traditional RAM
+            // We need to reconfigure for EEPROM usage
+
+#    if defined(MK66F18)
         kinetis_hsrun_disable();
         FTFL->FCCOB0 = FTFE_FCCOB0_CCOBn_SET(0x80);  // PGMPART = Program Partition Command
```

Comment:
> Why does moving the #if guards outside of the kinetis_hsrun_disable() function help with build error?

Because those functions are unused on other platforms, and warnings occur as a result... which are treated as errors.

Reply:
> I only changed FTFL->FCCOB0 = 0x80; to FTFL->FCCOB0 = FTFE_FCCOB0_CCOBn_SET(0x80); in PR #12947

FTFE_FCCOB0_CCOBn_SET does not exist on non-MK66F18 boards, and ends up failing the build.

Reply:
```
Compiling: tmk_core/common/chibios/eeprom_teensy.c                                                 tmk_core/common/chibios/eeprom_teensy.c:105:12: error: 'kinetis_hsrun_enable' defined but not used [-Werror=unused-function]
 static int kinetis_hsrun_enable(void) {
            ^~~~~~~~~~~~~~~~~~~~
tmk_core/common/chibios/eeprom_teensy.c:47:12: error: 'kinetis_hsrun_disable' defined but not used [-Werror=unused-function]
 static int kinetis_hsrun_disable(void) {
            ^~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```
...and:
```
Compiling: tmk_core/common/chibios/eeprom_teensy.c                                                 tmk_core/common/chibios/eeprom_teensy.c: In function 'eeprom_initialize':
tmk_core/common/chibios/eeprom_teensy.c:237:24: error: implicit declaration of function 'FTFE_FCCOB0_CCOBn_SET' [-Werror=implicit-function-declaration]
         FTFL->FCCOB0 = FTFE_FCCOB0_CCOBn_SET(0x80);  // PGMPART = Program Partition Command
                        ^~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

Reply:
> > I only changed FTFL->FCCOB0 = 0x80; to FTFL->FCCOB0 = FTFE_FCCOB0_CCOBn_SET(0x80); in PR #12947
> 
> FTFE_FCCOB0_CCOBn_SET does not exist on non-MK66F18 boards, and ends up failing the build.

Gotcha. We should just change it back to `FTFL->FCCOB0 = 0x80` then.

Reply:
>  FCCOB0 needs to be set outside of MK66F18, too, so this #if directive is too broad.
>
> I only changed FTFL->FCCOB0 = 0x80; to FTFL->FCCOB0 = FTFE_FCCOB0_CCOBn_SET(0x80); in PR #12947

Are we missing a header or something, then?

Reply:
The macro is defined in lib/chibios-contrib/os/common/ext/CMSIS/KINETIS/MK66F18.h, but not for other devices. Not sure why.

As i said, easiest to change it back to just `0x80`.